src/client/tab.rs

```diff
@@ -73,6 +77,7 @@ pub struct TabData {
     pub position: usize,
     pub name: String,
     pub active: bool,
+    pub input_mode: InputMode,
```

Comment:
Why do we need the InputMode on the TabData?

Reply:
oops, that should've been a part of a cleanup 😅